libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -180,6 +180,27 @@ void IRGeneratorForStatements::endVisit(VariableDeclarationStatement const& _var
 				declare(m_context.addLocalVariable(*decl));
 }
 
+bool IRGeneratorForStatements::visit(Conditional const& _conditional)
+{
+	_conditional.condition().accept(*this);
+
+	string condition = expressionAsType(_conditional.condition(), *TypeProvider::boolean());
+	auto type = _conditional.annotation().type;
+	IRVariable toAssign(m_context.newYulVariable(), *type);
+	declare(toAssign);
```

Comment:
Actually: after thinking about it once more:

You can actually skip the whole ``toAssign`` thing and just ``declare(_conditional)`` here and use ``assign(conditional, ...)`` below.

Reply:
yea true